pkg/scheduler/framework/plugins/podtopologyspread/scoring.go

```diff
@@ -135,6 +135,10 @@ func (pl *PodTopologySpread) PostFilter(
 			// <matchSum> indicates how many pods (on current node) match the <constraint>.
 			matchSum := int64(0)
 			for _, existingPod := range nodeInfo.Pods() {
+				// Bypass terminating Pod (see #87621).
+				if existingPod.DeletionTimestamp != nil || existingPod.Namespace != pod.Namespace {
```

Comment:
I'm curious about why the namespace is not checked before?

Reply:
I have to admit it was a pre-existing bug...

Reply:
Got it.

Reply:
Let's add a separate regression test for the namespace.